caddyhttp/httpserver/context.go

```diff
@@ -63,6 +63,18 @@ func (c Context) Header(name string) string {
 	return c.Req.Header.Get(name)
 }
 
+// Hostname gets the (remote) hostname of the client making the request.
+func (c Context) Hostname() string {
+	ip := c.IP()
+
+	address, err := net.LookupAddr(ip)
+	if err != nil {
+		return c.Req.RemoteAddr
+	}
+
+	return address[0]
```

Comment:
Yeah, I did a bunch of digging and I couldn't find an appropriate example proving that it would never return empty. Just being a bit paranoid just in case!